site/tutorials/snippets/08/self-healing/upscalePrometheus.md

```diff
@@ -0,0 +1,162 @@
+
+Negative
+: First, make sure you are in the correct folder `examples/onboarding-carts` otherwise the next commands will fail.
+
+For this usecase, we have prepared another version of the SLI file. Add it with the following command:
+<!-- command -->
+```
+keptn add-resource --project=sockshop --stage=production --service=carts --resource=sli-config-prometheus-selfhealing.yaml --resourceUri=prometheus/sli.yaml 
+```
+
+Add the prepared SLO file for self-healing to the production stage using the Keptn CLIs add-resource command:
+
+<!-- command -->
+```
+keptn add-resource --project=sockshop --stage=production --service=carts --resource=slo-self-healing.yaml --resourceUri=slo.yaml
+```
+
+Note: The SLO file contains an objective for response_time_p90.
+
+
+Configure Prometheus with the Keptn CLI (this configures the [Alert Manager](https://prometheus.io/docs/alerting/configuration/) based on the slo.yaml file):
+
+<!-- command -->
+```
+keptn configure monitoring prometheus --project=sockshop --service=carts
+```
+
+Configure remediation actions for up-scaling based on Prometheus alerts:
+
+<!-- command -->
+```
+keptn add-resource --project=sockshop --stage=production --service=carts --resource=remediation.yaml --resourceUri=remediation.yaml
+```
+
+This is the content of the file that has being added:
+
+```
+apiVersion: spec.keptn.sh/0.1.4
+kind: Remediation
+metadata:
+  name: carts-remediation
+spec:
+  remediations:
+    - problemType: Response time degradation
+      actionsOnOpen:
+        - action: scaling
+          name: scaling
+          description: Scale up
+          value: 1
+    - problemType: response_time_p90
+      actionsOnOpen:
+        - action: scaling
+          name: scaling
+          description: Scale up
+          value: 1
+```
+
+## Generate load for the service
+Duration: 3:00
+
+To simulate user traffic that is causing an unhealthy behavior in the carts service, please execute the following script. This will add special items into the shopping cart that cause some extensive calculation.
+
+1. Move to the correct folder for the load generation scripts:
+
+    <!-- command -->
+    ```
+    cd ../load-generation/cartsloadgen/deploy
+    ```
+
+1. Start the load generation script: 
+
+    <!-- command -->
+    ```
+    kubectl apply -f cartsloadgen-faulty.yaml
+    ```
+
+1. (optional:) Verify the load in Prometheus.
+    - Make a port forward to access Prometheus:
+
+    ```
+    kubectl port-forward svc/prometheus-service -n monitoring 8080:8080
+    ```
+    
+    - Access Prometheus from your browser on [http://localhost:8080](http://localhost:8080).
+
+    - In the **Graph** tab, add the expression 
+
+    ```
+    histogram_quantile(0.9, sum by(le) (rate(http_response_time_milliseconds_bucket{job="carts-sockshop-production-primary"}[3m])))
```

Comment:
@bacherfl not sure if this is correct, but it was
```
histogram_quantile(0.9, sum by(le) (rate(http_response_time_milliseconds_bucket{job="carts-sockshop-production"}[3m])))
```
and I added the `-primary` behind. Without this I did not get the plotted metric.